tests/reduction/reduction_get_lambda.h

```diff
@@ -0,0 +1,269 @@
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] &= accessor[idx];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::bit_xor<VariableT>>::value) {
+    return [=](sycl::id<1> idx, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] ^= accessor[idx];
+      }
+    };
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::nd_range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
+ *  @tparam AccessorT buffer accessor type
+ *  @param accessor Accessor to the buffer
+ *  @param number_elements Number elements in sycl::span, used only when
+ *         constructing reducer with span
+ *  @retval Lambda with chosen operator
+ */
+template <typename VariableT, typename FunctorT, typename UseCombineFlagT,
+          typename AccessorT>
+auto get_lambda_with_nd_range_for_span(AccessorT accessor,
+                                       size_t number_elements) {
+  if constexpr (UseCombineFlagT::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i].combine(accessor[nd_item.get_global_id()]);
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::multiplies<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] *= accessor[nd_item.get_global_id()];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT, sycl::plus<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] += accessor[nd_item.get_global_id()];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT, sycl::bit_or<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] |= accessor[nd_item.get_global_id()];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::bit_and<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] &= accessor[nd_item.get_global_id()];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::bit_xor<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] ^= accessor[nd_item.get_global_id()];
+      }
+    };
+  }
```

Comment:
What I mean is that all this function could be written as something like
```c++
return [=](sycl::nd_item<1> nd_item, auto &reducer) {
      for (size_t i = 0; i < number_elements; i++) {
        operation_dispatch<FunctorT>(reducer[i], accessor[nd_item.get_global_id()]);
      }
    };
```
Computer science is about laziness as a zen art of life. :-)

Reply:
If I correctly understand You
I've done this in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/165/commits/823e2fbe3bded9166662dc948031fd61294f4e36